### PR TITLE
🌱 Descriptive Upgrade Constraint Policy constants

### LINKED ISSUE #1151: Update the allowed values of the `ClusterExtension.Spec.UpgradeConstraintPolicy` to be more descriptive
For the `ClusterExtension.Spec.UpgradeConstraintPolicy` field, the “Enforce” and “Ignore” allowed values are not very descriptive of their behavior. In an effort to be more descriptive it is proposed that the following changes are made to the allowed value names:
- `Enforce` → `CatalogProvided`
- `Ignore` → `SelfCertified`

The GoDoc comment should be updated to include explicit information as to what each allowed value means when specified.

### PR DESCRIPTION
**Pull request by skattoju**
Fixes #1151 

This PR renames the constant values used in the API for UpgradeConstraintPolicy and updates their references in the documentation.

### CODE REVIEW

File: api/v1alpha1/clusterextension_types.go [hunk 1]
```diff
@@ -337,8 +337,8 @@ type CatalogSource struct {
 	//
 	// If unspecified, the default value is "Enforce".
 	//
-	//+kubebuilder:validation:Enum:=Enforce;Ignore
-	//+kubebuilder:default:=Enforce
+	//+kubebuilder:validation:Enum:=CatalogProvided;SelfCertified
```
Comment (everettraven) [inline api/v1alpha1/clusterextension_types.go:340]: We should also update the values in the GoDoc for this field

Comment (skattoju) [reply api/v1alpha1/clusterextension_types.go:340]: Updated, thanks for pointing it out 👍

**Review by LalatenduMohanty — DISMISSED** [review]
/lgtm

**Comment by LalatenduMohanty** [comment]
Conflicts because PR https://github.com/operator-framework/operator-controller/pull/1237 has merged

**Review by everettraven — DISMISSED** [review]

**Comment by everettraven** [comment]
Approved and then looked at CI - looks like you need to:
- rebase and resolve merge conflicts
- regenerate manifests with `make manifests`

**Comment by skattoju** [comment]
I've rebased and resolved conflicts. The CI failures are because we are changing the API right ?

**Comment by everettraven** [comment]
@skattoju It looks like there is still a sanity failure. I _do_ expect the go-apidiff failure though.

**Comment by everettraven** [comment]
Looks like the sanity failure is due to the newly added types documentation

**Comment by skattoju** [comment]
looks like sanity is passing after doing `make verify` and committing changes to `docs/refs/api/operator-controller-api-reference.md` 👍

File: internal/resolve/catalog_test.go [hunk 2]
```diff
@@ -673,7 +673,7 @@ func (w staticCatalogWalker) WalkCatalogs(ctx context.Context, _ string, f Catal
 			ObjectMeta: metav1.ObjectMeta{
 				Name: k,
 				Labels: map[string]string{
-					"olm.operatorframework.io/metadata.name": k,
+					"olm.operatorframework.io/name": k,
```
Comment (joelanford) [inline internal/resolve/catalog_test.go:676]: Looks like a rebase mistake?

The correct label name is `olm.operatorframework.io/metadata.name` (we changed it recently)

File: internal/resolve/catalog_test.go [hunk 3]
```diff
@@ -851,8 +851,8 @@ func TestClusterExtensionMatchLabel(t *testing.T) {
 		},
 	}
 	r := CatalogResolver{WalkCatalogsFunc: w.WalkCatalogs}
-	ce := buildFooClusterExtension(pkgName, []string{}, "", ocv1alpha1.UpgradeConstraintPolicyEnforce)
-	ce.Spec.Source.Catalog.Selector.MatchLabels = map[string]string{"olm.operatorframework.io/metadata.name": "b"}
+	ce := buildFooClusterExtension(pkgName, []string{}, "", ocv1alpha1.UpgradeConstraintPolicyCatalogProvided)
+	ce.Spec.Source.Catalog.Selector.MatchLabels = map[string]string{"olm.operatorframework.io/name": "b"}
```
Comment (joelanford) [inline internal/resolve/catalog_test.go:855]:
```suggestion
	ce.Spec.Source.Catalog.Selector.MatchLabels = map[string]string{"olm.operatorframework.io/metadata.name": "b"}
```

File: internal/resolve/catalog_test.go (hunk 2 above)
Comment (joelanford) [inline internal/resolve/catalog_test.go:676]:
```suggestion
					"olm.operatorframework.io/metadata.name": k,
```

File: internal/resolve/catalog_test.go [hunk 4]
```diff
@@ -870,8 +870,8 @@ func TestClusterExtensionNoMatchLabel(t *testing.T) {
 		},
 	}
 	r := CatalogResolver{WalkCatalogsFunc: w.WalkCatalogs}
-	ce := buildFooClusterExtension(pkgName, []string{}, "", ocv1alpha1.UpgradeConstraintPolicyEnforce)
-	ce.Spec.Source.Catalog.Selector.MatchLabels = map[string]string{"olm.operatorframework.io/metadata.name": "a"}
+	ce := buildFooClusterExtension(pkgName, []string{}, "", ocv1alpha1.UpgradeConstraintPolicyCatalogProvided)
+	ce.Spec.Source.Catalog.Selector.MatchLabels = map[string]string{"olm.operatorframework.io/name": "a"}
```
Comment (joelanford) [inline internal/resolve/catalog_test.go:874]:
```suggestion
	ce.Spec.Source.Catalog.Selector.MatchLabels = map[string]string{"olm.operatorframework.io/metadata.name": "a"}
```

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Pull request merged**